src/Ocelot/DownstreamRouteFinder/UrlMatcher/UrlPathPlaceholderNameAndValueFinder.cs

```diff
@@ -50,11 +50,16 @@ public Response<List<PlaceholderNameAndValue>> Find(string path, string query, s
 
                     return new OkResponse<List<PlaceholderNameAndValue>>(placeHolderNameAndValues);
                 }
-                else if (IsCatchAll(path, counterForPath, pathTemplate))
+                else if (IsCatchAll(path, counterForPath, pathTemplate) || (pathTemplate[counterForTemplate] == '{') && NoMoreForwardSlash(pathTemplate, counterForTemplate) && NotPassedQueryString(pathTemplate, pathTemplate.Length))
```

Comment:
Why did you add this long and complex condition here just for user cases?

Which paths will be handled here?

Reply:
This Logic here is the one that will handle the case of a catch all in the end of line and handles it with the same logic as the normal catchall